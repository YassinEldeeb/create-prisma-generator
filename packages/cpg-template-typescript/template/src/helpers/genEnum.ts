import { DMMF } from '@prisma/generator-helper'

export const genEnum = ({ name, values }: DMMF.DatamodelEnum) => {
  const enumValues = values.map(({ name }) => `${name}="${name}"`).join(',\n')

  return `enum ${name} { \n${enumValues}\n }`
}
