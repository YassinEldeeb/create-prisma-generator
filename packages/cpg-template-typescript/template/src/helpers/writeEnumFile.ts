import path from 'path'
import { writeFileSafely } from '../utils/writeFileSafely'

export const writeEnumFile = async (
  enumName: string,
  tsEnum: string,
  outputPath: string,
) => {
  const writeLocation = path.join(outputPath, `${enumName}.ts`)
  await writeFileSafely(writeLocation, tsEnum)
}
