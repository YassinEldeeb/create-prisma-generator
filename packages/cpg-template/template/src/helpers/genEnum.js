export const genEnum = ({ name, values }) => {
  const enumValues = values.map(({ name }) => `${name}="${name}"`).join(',\n')

  return `enum ${name} { \n${enumValues}\n }`
}
