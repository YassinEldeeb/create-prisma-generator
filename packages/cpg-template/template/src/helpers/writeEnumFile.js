import path from 'path'
import { writeFileSafely } from '../utils/writeFileSafely'

export const writeEnumFile = async (
  enumName,
  tsEnum,
  outputPath,
) => {
  const writeLocation = path.join(outputPath, `${enumName}.ts`)
  await writeFileSafely(writeLocation, tsEnum)
}
