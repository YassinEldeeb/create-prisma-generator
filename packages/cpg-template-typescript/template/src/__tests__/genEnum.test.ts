import { DMMF } from '@prisma/generator-helper'
import { genEnum } from '../helpers/genEnum'
import { getSampleDMMF } from './__fixures__/writeSampleDMMF'

let sampleDMMF: DMMF.Document
beforeAll(async () => {
  sampleDMMF = await getSampleDMMF()
})

test('enum generation', () => {
  sampleDMMF.datamodel.enums.forEach((enumInfo) => {
    expect(genEnum(enumInfo)).toMatchSnapshot(enumInfo.name)
  })
})
