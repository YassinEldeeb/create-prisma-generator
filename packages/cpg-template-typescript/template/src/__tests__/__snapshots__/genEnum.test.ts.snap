// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enum generation: Language 1`] = `
"enum Language { 
Typescript=\\"Typescript\\",
Javascript=\\"Javascript\\",
Rust=\\"Rust\\",
Go=\\"Go\\",
Python=\\"Python\\",
Cpp=\\"Cpp\\"
 }"
`;

exports[`enum generation: NotificationType 1`] = `
"enum NotificationType { 
newPosts=\\"newPosts\\",
newComments=\\"newComments\\",
newFollowers=\\"newFollowers\\",
reply=\\"reply\\",
heartOnPost=\\"heartOnPost\\",
heartOnComment=\\"heartOnComment\\",
heartOnReply=\\"heartOnReply\\"
 }"
`;

exports[`enum generation: State 1`] = `
"enum State { 
Active=\\"Active\\",
Pending=\\"Pending\\",
Banned=\\"Banned\\"
 }"
`;
